Update the `repr(transparent)` section to reflect the current state (#376)
rust-lang/nomicon: src/other-reprs.md

## repr(transparent)

> This can only be used on structs with a single non-zero-sized field
> or enums with only one non-zero-sized field (there may be additional zero-sized fields).

Reviewer: I think it might be good to be explicit that the enum can only have a single variant.  I might also rework this a little to avoid the duplication, so maybe something like:

```suggestion
`#[repr(transparent)]` can only be used on a struct or single-variant enum that has a single non-zero-sized field (there may be additional zero-sized fields).
```
